### DISCUSSION TITLE
FYI to terraform helm provider users when Airflow Jobs are not created

### DISCUSSION THREAD

**Discussion by EliMor** [apache/airflow#17677] [General]
Hi all.

I was trying to wrap the Airflow helm chart using the terraform helm provider: 
https://github.com/hashicorp/terraform-provider-helm
https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release

But spent a while scratching my head why none of the Airflow Job pods were getting created despite it working just fine with the values.yaml file I have and installing it manually via the helm cli. 

Just an FYI to anyone seeking this path or encountering someone with this complaint, here's the issue to track;
https://github.com/hashicorp/terraform-provider-helm/issues/683

TLDR:
Try setting
```  wait = false ``` in your terraform file for now: [ref](https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#wait)

I hope this saves someone a little bit of time.

**Comment by potiuk** [comment]
@jedcunningham @kaxil  -> I think we need to rethink the migration as post-install, It does create problems for people so maybe we should turn it into  "init-containers" approach? 

The "migration lock" mechanism that we have in 2.2 should help with it, but maybe we can work out something else for pre 2.2?

I particularly do not like the solution with continuosly running migration job, but possibly just migration init container for the scheduler could to the trick?  

My thinking is that we can potentially implement a very simple python script (that could be hard-coded in the chart not airlfow) that could actively monitor if the migration is completed for the "currently expected" version of sqlalchemy and run it for all other Pods as init containers and only run the migration by the "first" scheduler's init container. That all seems to be doable at the chart level, without modifying airflow and it could work for all released version of Airflow 2.

WDYT?

**Comment by vara-bonthu** [comment]
@EliMor  @potiuk I am hitting the similar issue with latest Airflow Helm Chart(`Release version 1.6.0)`.  I am able to deploy the Airflow on EKS with Helm Chart CLI successfully however, the same deployment `values.yaml` failing with **Terraform Helm Release**.

I noticed that `airflow-run-airflow-migrations-xyz` Kubernetes job is not being triggered when i deploy with **Terraform Helm release** however the same works with Helm CLI. All the pods were failing with `wait for migrations` when i deploy with Terraform.

It does look like `wait = false` worked for me. This issue seems to be an intermittent with Terraform. I still need to run few more tests with fresh install and upgrades as well to see if Terraform Helm is stable to deploy

Here is the Terraform Airflow deployment example on `Amazon EKS with Postgres RDS` https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/878/. This example has been implemented with all the Airflow production best practices. Looking forward to hear the feedback from the Airflow experts 👍🏼

**Reply by potiuk** [reply]
I think this is more of a question to @jedcunningham and @dstandish  and @ephraimbuddy  (last time they were mostlty involved in that 
 part :). Whether our migration strategy with migration job is going to change. I believe the current one has some rough edges especially when non pure helm installation is used (ArgoCD, Terraform and few others) and while there are some workarounds (like you mentioned wait=False and disabling migration hooks). I know we at least discussed that in some future backwards-incompatible Helm Chart version we might change it, but I am not sure where we are with it now :)
